lib/internal/modules/esm/hooks.js

```diff
@@ -164,6 +167,17 @@ class Hooks {
    * @returns {any | Promise<any>} User data, ignored unless it's a promise, in which case it will be awaited.
    */
   addCustomLoader(url, exports, data) {
+    const alreadyKnown = ArrayPrototypeSome(ArrayPrototypeMap(['initialize', 'resolve', 'load'], (hookName) => {
+      if (this.#chains[hookName]) {
+        return ArrayPrototypeFilter(
+          this.#chains[hookName], (el) => el.url === url && el.data === data).length === 1;
```

Comment:
I'm not sure this comparison of data is correct and this might actually never be triggered. Can you add a test?

Reply:
I think one good test would be (all filenames for illustration only, please use whatever names correspond to the appropriate fixtures):
- Node is run via `node --import=register.js app.js`
- `register.js` contains a `register` call that registers some hooks, and one of these hooks prints something on initialization
- `app.js` contains `new Worker('./worker.js')` to create a worker thread without any specific customization (no `execArgv`)
- Verify that the “print on initialization” doesn’t happen a second time

The point of this test is to ensure that even though `new Worker` without `execArgv` inherits the `--import` flag from the initial Node process, and even though `register.js` runs twice, the hooks don’t get registered twice. This filter check prevents double registration of the same hooks.

Reply:
> I'm not sure this comparison of data is correct and this might actually never be triggered. Can you add a test?

Of course. This must be a deep equality compare. Will update. Tests need generally to be added for quite a few things. Just wanted to make sure there is agreement on the approach and featureset/constraints before.

Reply:
@dygabo In my (soon to be published) PR I used `isDeepStrictEqual` from `internal/util/comparisons`, which is the one internally used from `assert`.